readme.md

```diff
@@ -245,12 +244,130 @@ interface Message {
     /** ID of the message this message responds to */
     inReplyTo?: string
 
-    /** when this message B is a response to the message A,
+    /** A list of message IDs of previous messages. When this message B is a response to the message A,
      *  B.references = [...A.references, A.inReplyTo] */
     references?: string[]
 }
 ```
 
+#### Quote
+
+```javascript
+interface CostBreakdown {
+    tax: Record<string, unknown>
+    net: number
+    gross: number
+}
+
+interface Quote {
+    cost: CostBreakdown
+    currency: string
+    timeframe: string
+    termsAndConditions: string
+}
+```
+
+#### Proposed Verifiable Credential
+
+The `ProposedCredential` is a partial `KiltCredentialV1`.
+The attester proposes the content of structure of the credential.
+Since the attester might not yet know the subjects DID, the `credentialSubject.id` property is optional.
+The credential is also not attested, therefore the `proof` and `issuanceDate` properties are missing.
+
+```javascript
+interface ProposedCredential {
+    "@context": [
+        "https://www.w3.org/2018/credentials/v1",
+        "https://www.kilt.io/contexts/credentials"
+    ],
+    type: [
+        "VerifiableCredential",
+        "KiltCredentialV1",
+        "kilt:ctype:0xf0fd09f9ed6233b2627d37eb5d6c528345e8945e0b610e70997ed470728b2ebf"
+    ],
+    id: string,
+    nonTransferable: true,
+    credentialSubject: {
+        "@context": {
+            "@vocab": "kilt:ctype:0xf0fd09f9ed6233b2627d37eb5d6c528345e8945e0b610e70997ed470728b2ebf#"
+        },
+        id?: string,
+        [key: string]: any
+    },
+    issuer: string,
+    federatedTrustModel: FederatedTrust[],
+    credentialStatus: CredentialStatus,
+    credentialSchema: {
+        id: "ipfs://QmRpbcBsAPLCKUZSNncPiMxtVfM33UBmudaCMQV9K3FD5z",
+        type: "JsonSchema2023"
+    },
+}
+```
+
+#### Agreed Verifiable Credential
+
+The `AgreedCredential` is a partial `KiltCredentialV1`.
+It previews the credential that both the claimer and attester agreed upon.
+Since the `AgreedCredential` is not attested, it doesn't contain the `issuanceDate` and only a partial `proof`.
+The `proof` contains the `type` which is currently limited to `KiltAttestationProofV1` and the `salt` and `commitments`.
+Note that the credential subject MUST be present.
+
+```javascript
```

Comment:
js does not have a type system
```suggestion
```typescript
```

Reply:
> js does not have a type system

not  sure if i can agree with that statement. JS has types so i would argue that it  has a  type  system.  Anyways I changed  it  to typescript